src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -1,545 +1,163 @@
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
+    const customWorkerResponses = _.merge(
+      _.cloneDeep(mockWorkerResponses),
+      { 'loading_gene_id-expression': () => stalledResponse },
     );
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
+    seekFromAPI.mockClear();
+    seekFromAPI.mockImplementation((a, b, c, requested) => customWorkerResponses[requested]());
 
-    expect(loadMarkerGenes).not.toHaveBeenCalled();
-  });
+    await loadAndRenderDefaultHeatmap();
 
-  it('shows error if marker genes couldn\'t be loaded', async () => {
-    const store = mockStore({
-      ...initialState,
-      networkResources: { environment: Environment.DEVELOPMENT },
-      cellSets: {
-        ...initialState.cellSets,
-        loading: true,
-        error: false,
-      },
-      genes: {
-        ...initialState.genes,
-        markers: {
-          ...initialState.genes.markers,
-          loading: false,
-          error: 'some marker error, probably something with parsing as always',
-        },
-      },
-      experimentSettings: {
-        ...initialState.experimentSettings,
-        processing: {
-          ...initialState.processing,
-          configureEmbedding: {
-            clusteringSettings: { methodSettings: { louvain: { resolution: 10 } } },
-          },
-        },
-      },
+    // Renders correctly
+    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();
+
+    // If a new gene suddenly is suddenly selected
+    await act(async () => {
+      storeState.dispatch(loadGeneExpression(experimentId, [...markerGenesData5.order, 'loading_gene_id'], 'interactiveHeatmap'));
     });
 
-    const louvainClusters = _.range(60).map((clusterIndex) => ({ key: `louvain-${clusterIndex}` }));
-    getCellSetsHierarchyByKeys.mockReturnValue(() => (
-      [{
-        key: 'louvain',
-        name: 'louvain clusters',
-        type: 'cellSets',
-        children: louvainClusters,
-      }]));
-
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
-    );
+    // Loading screen shows up
+    expect(screen.getByText(/We're getting your data.../i)).toBeInTheDocument();
+  });
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
-    expect(loadMarkerGenes).not.toHaveBeenCalled();
+  // Add test for expressionDataError
 
-    // Shows error component
-    expect(component.find('PlatformError').length).toEqual(1);
+  // Add test for viewError
 
-    //
-    await act(async () => {
-      component.find('PlatformError').props().onClick();
-    });
+  // Add test for markerGenesLoadingError
 
-    // on clicking platform error button it reloads
-    const expectedMarkerGenes = 3;
-    expect(loadMarkerGenes).toHaveBeenCalledWith('123', 10, 'interactiveHeatmap', expectedMarkerGenes);
-  });
+  // Think more tests
```

Comment:
Can you test for:
1. If cells are hidden from embedding, they also get hidden from the heatmap. Same for unhiding.
2. If user selects to view expression of gene from the gene list, the heatmap updates correctly.
3. The heatmap tooltips work.
4. The settings on the heatmap work -- when you click and unclick stuff there the heatmap updates accordingly.

Reply:
1. Impemented.
2. Implemented.
3. Implemented.
4. We just receive the data from a `useSelector`, but I implemented a check that the heatmap responds to that data changing.